rush-plugins/rush-amazon-s3-build-cache-plugin/src/AmazonS3Client.ts

```diff
@@ -103,27 +134,65 @@ export class AmazonS3Client {
 
   public async getObjectAsync(objectName: string): Promise<Buffer | undefined> {
     this._writeDebugLine('Reading object from S3');
-    const response: fetch.Response = await this._makeRequestAsync('GET', objectName);
-    if (response.ok) {
-      return await response.buffer();
-    } else if (response.status === 404) {
-      return undefined;
-    } else if (response.status === 403 && !this._credentials) {
-      return undefined;
-    } else {
-      this._throwS3Error(response, await this._safeReadResponseText(response));
-    }
+    type Response = RetriableRequestResponse<Buffer | undefined>;
+    const sendRequest: () => Promise<Response> = async (): Promise<Response> => {
+      const response: fetch.Response = await this._makeRequestAsync('GET', objectName);
+      if (response.ok) {
+        return {
+          hasNetworkError: false,
+          response: await response.buffer()
+        };
+      } else if (response.status === 404) {
+        return {
+          hasNetworkError: false,
+          response: undefined
+        };
+      } else if (
+        (response.status === 400 || response.status === 401 || response.status === 403) &&
+        !this._credentials
+      ) {
+        // unauthorized due to not providing credentials,
+        // silence error for better DX when e.g. running locally without credentials
+        return {
+          hasNetworkError: false,
+          response: undefined
+        };
```

Comment:
If there are no credentials and we get a 401, or if we get a 403, we should tell the user that they need to authenticate. This looks the same as a cache miss, so a user with missing, expired, or insufficient credentials will never get entries from cache and will never know that they need to do something about that.